src/Expr.cc

```diff
@@ -2382,6 +2382,17 @@ IndexExpr::IndexExpr(ExprPtr arg_op1, ListExprPtr arg_op2, bool arg_is_slice, bo
     if ( IsError() )
         return;
 
+    if ( op1->GetType()->Tag() == TYPE_TABLE ) { // Check for a table[pattern] being indexed by a string
+        const auto& table_type = op1->GetType()->AsTableType();
+        const auto& rhs_type = op2->GetType()->AsTypeList()->GetTypes();
+        if ( table_type->IsPatternIndex() && table_type->Yield() && rhs_type.size() == 1 &&
+             IsString(rhs_type[0]->Tag()) ) {
+            is_pattern_table = true;
+            SetType(make_intrusive<VectorType>(op1->GetType()->Yield()));
```

Comment:
Vector preferable over set?

Reply:
The plus of a vector is that if the developer knows there can only be one match, they can get to it with `[0]`, whereas we don't have a convenient "give me an arbitrary set element" operator.

Reply:
Also cheaper as we don't need to serialize values into and from a set, allows to mutate the values in case of non-atomic yield types. Non-lossy, too, if there's a use-case for the same value for different patterns - summing up scores of matched patterns?